src/modules/auctions/contractLoader.ts

```diff
@@ -0,0 +1,41 @@
+import { AbiRegistry, SmartContract, Address } from '@multiversx/sdk-core';
+import { Logger } from '@nestjs/common';
+import * as fs from 'fs';
+
+export class ContractLoader {
+  private readonly abiPath: string;
+  private abi: AbiRegistry | undefined = undefined;
+  private readonly logger: Logger;
+
+  constructor(abiPath: string) {
+    this.abiPath = abiPath;
+    this.logger = new Logger(ContractLoader.name);
+  }
+
+  private async load(): Promise<AbiRegistry> {
+    try {
+      const jsonContent: string = await fs.promises.readFile(this.abiPath, { encoding: 'utf8' });
+      const json = JSON.parse(jsonContent);
+
+      const abiRegistry = AbiRegistry.create(json);
+
+      return abiRegistry;
+    } catch (error) {
+      this.logger.log(`Unexpected error when trying to create smart contract from abi`);
+      this.logger.error(error);
+
+      throw new Error('Error when creating contract from abi');
+    }
+  }
+
+  async getContract(contractAddress: string): Promise<SmartContract> {
```

Comment:
`getContract()` can be moved at top, after constructor, before `load()`.

Reply:
done